pkg/resources/provisioning.cattle.io/v1/cluster/mutator_test.go

```diff
@@ -617,3 +618,96 @@ func TestAdmitPreserveUnknownFields(t *testing.T) {
 	assert.Nil(t, response.Patch)
 
 }
+
+func TestDynamicSchemaDrop(t *testing.T) {
+	cluster := &v1.Cluster{
+		Spec: v1.ClusterSpec{
+			RKEConfig: &v1.RKEConfig{
+				MachinePools: []v1.RKEMachinePool{
+					{
+						Name: "A",
+					},
+				},
+			},
+		},
+	}
+	raw, err := json.Marshal(cluster)
+	assert.Nil(t, err)
+
+	m := ProvisioningClusterMutator{}
+
+	request := &admission.Request{
+		AdmissionRequest: admissionv1.AdmissionRequest{
+			Object: runtime.RawExtension{
+				Raw: raw,
+			},
+			OldObject: runtime.RawExtension{
+				Raw: raw,
+			},
+		},
+	}
+
+	// Always allowed on Create
+	request.Operation = admissionv1.Create
+	output := cluster.DeepCopy()
+	response, err := m.handleDynamicSchemaDrop(request, output)
+	assert.Nil(t, err)
+	assert.True(t, response.Allowed)
+	assert.True(t, reflect.DeepEqual(cluster, output))
+
+	// Allowed on update if no schema present
+	request.Operation = admissionv1.Update
+	output = cluster.DeepCopy()
+	response, err = m.handleDynamicSchemaDrop(request, output)
+	assert.Nil(t, err)
+	assert.True(t, response.Allowed)
+	assert.True(t, reflect.DeepEqual(cluster, output))
+
+	// Allowed on update if schemas match
+	cluster.Spec.RKEConfig.MachinePools[0].DynamicSchemaSpec = "a"
+	raw, err = json.Marshal(cluster)
+	assert.Nil(t, err)
+	request.AdmissionRequest.OldObject.Raw = raw
+	request.AdmissionRequest.Object.Raw = raw
+	output = cluster.DeepCopy()
+	response, err = m.handleDynamicSchemaDrop(request, output)
+	assert.Nil(t, err)
+	assert.True(t, response.Allowed)
+	assert.True(t, reflect.DeepEqual(cluster, output))
+
+	// Allowed on update if new pool is added
+	cluster.Spec.RKEConfig.MachinePools = append(cluster.Spec.RKEConfig.MachinePools, v1.RKEMachinePool{Name: "B"})
+	raw, err = json.Marshal(cluster)
+	assert.Nil(t, err)
+	request.AdmissionRequest.Object.Raw = raw
+	output = cluster.DeepCopy()
+	response, err = m.handleDynamicSchemaDrop(request, output)
+	assert.Nil(t, err)
+	assert.True(t, response.Allowed)
+	assert.True(t, reflect.DeepEqual(cluster, output))
+
+	// Rejected on update if schema is removed
```

Comment:
Nit: This comment is a bit inaccurate - the request is always allowed, but we mutate it under select circumstances.
```suggestion
	// Allowed on updated if schema is removed, but old SchemaSpec is re-introduced.
```